web/src/main/java/org/openmrs/layout/web/address/AddressSupport.java

```diff
@@ -29,6 +29,8 @@
 @Deprecated
 public class AddressSupport extends LayoutSupport<AddressTemplate> implements GlobalPropertyListener {
 	
```

Comment:
To explain at a high level, there is not really a 'correct' class name for AddressTemplate as you are assuming. There is the 'api' package AddressSupport/Template code and the 'web' package AddressSupport/Template code. When old modules are running, both sets of code need to run. Both AddressSupport classes need to instantiate AddressTemplates in the matching package. That can't be done if there is a single class defined in the db.

At a detailed level, just look at line 147 of 'web' AddressSupport. Of course it cannot cast an 'API' AddressTemplate in the db to a 'web' AddressTemplate. Even if you were to try to fix 'web' AddressSupport' to compile against 'API' AddressTemplate, module code would bomb that expects a 'web' AddressSupport to provide a 'web' AddressTemplate. 

My junit test demonstrates the failure before the fix, and passes after the fix (yes I am a classically-trained TDD developer :-)

The NameSupport problem is a little different in detail but similar in concept. 2 sets of code need to run from 2 packages.




> On Mar 26, 2015, at 12:38 PM, Wyclif Luyima <notifications@github.com> wrote:
> 
> In web/src/main/java/org/openmrs/layout/web/address/AddressSupport.java:
> 
> > @@ -29,6 +29,8 @@
> >  @Deprecated
> >  public class AddressSupport extends LayoutSupport<AddressTemplate> implements GlobalPropertyListener {
> >  	
> I still don't understand why these changes are necessary. We expect the value from the DB to match the correct class name, if not then we need to fix the upgrade scripts to do so
> 
> —
> Reply to this email directly or view it on GitHub.
>